/* Patch created by @Moe-Baker from Github
 * Link: https://github.com/nillerusr/source-thirdparty/pull/4
 * Description: Modify SDL gamepad detection to ignore Android fingerprint sensors
 * Date: 2023-01-07
 */

--- SDL_gamecontroller.c	2024-12-24 14:37:19.224394057 +0100
+++ SDL_gamecontroller_patch.c	2024-12-24 14:37:03.036511967 +0100
@@ -1786,6 +1786,19 @@
     Uint16 version;
     Uint32 vidpid;
 
+    if (name)
+    {
+        if (SDL_strcmp(name, "uinput-fpc") == 0) {
+            /* The Google Pixel fingerprint sensor reports itself as a joystick */
+            return SDL_TRUE;
+        }
+
+        if (SDL_strcmp(name, "uinput-goodix") == 0) {
+            /* The xiamoi fingerprint sensor reports itself as a joystick */
+            return SDL_TRUE;
+        }
+    }
+
 #if defined(__LINUX__)
     if (name && SDL_strstr(name, "Motion Sensors")) {
         /* Don't treat the PS3 and PS4 motion controls as a separate game controller */
